types/aqb/index.d.ts

```diff
@@ -0,0 +1,1753 @@
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): InsertExpression;
+    in: InsertRet["into"];
+    in_: InsertRet["into"];
+}
+interface UpdateRetWithRet {
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): UpdateExpression;
+    in: UpdateRetWithRet["into"];
+    in_: UpdateRetWithRet["into"];
+}
+interface ReplaceRetWithRet {
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): ReplaceExpression;
+    in: ReplaceRetWithRet["into"];
+    in_: ReplaceRetWithRet["into"];
+}
+interface RemoveRet {
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): RemoveExpression;
+    in: RemoveRet["into"];
+    in_: RemoveRet["into"];
+}
+interface UpsertRet {
+    /**
+     *
+     * @param insertExpr
+     * @return
+     */
+    insert(insertExpr: any): UpsertRetInsertRet;
+}
+interface UpsertRetInsertRet {
+    /**
+     *
+     * @param updateOrReplaceExpr
+     * @return
+     */
+    update(updateOrReplaceExpr: any): UpsertRetInsertRetUpdateRet;
+    replace: UpsertRetInsertRet["update"];
+}
+interface UpsertRetInsertRetUpdateRet {
+    /**
+     *
+     * @param inCollection
+     * @return
+     */
+    into(inCollection: any): UpsertExpression;
+    in: UpsertRetInsertRetUpdateRet["into"];
+    in_: UpsertRetInsertRetUpdateRet["into"];
+}
+interface InsertRet {
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): InsertExpression;
+    in: InsertRet["into"];
+    in_: InsertRet["into"];
+}
+interface UpdateRetWithRet {
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): UpdateExpression;
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    with(collection: any): UpdateRetWithRet;
+    in: UpdateRetWithRet["into"];
+    in_: UpdateRetWithRet["into"];
+}
+interface ReplaceRetWithRet {
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    into(collection: any): ReplaceExpression;
+    /**
+     *
+     * @param collection
+     * @return
+     */
+    with(collection: any): ReplaceRetWithRet;
+    in: ReplaceRetWithRet["into"];
+    in_: ReplaceRetWithRet["into"];
+}
+/**
+ *
+ * @param self
+ * @param args
+ * @return
+ */
+declare function toArray(self: Expression, ...args: any[]): any[];
+/**
+ *
+ * @param str
+ * @return
+ */
+declare function isQuotedString(str: string): boolean;
+/**
+ *
+ * @param expr
+ * @return
+ */
+declare function wrapAQL(expr: Keyword): string;
+/**
+ *
+ * @param number
+ * @return
+ */
+declare function isValidNumber(number: number): boolean;
+/**
+ *
+ * @param number
+ * @return
+ */
+declare function castNumber(number: any): IntegerLiteral;
+/**
+ *
+ * @param bool
+ * @return
+ */
+declare function castBoolean(bool: any): BooleanLiteral;
+/**
+ *
+ * @param str
+ * @return
+ */
+declare function castString(str: any): number;
```

Comment:
Is it right for a function named `castString` to return `number`? I'm not familiar with the library, but it feels suspect.

Reply:
Lol it is very suspect... Checked out the lib... It's meant to return 
```js
SimpleReference|Identifier|RangeExpression|StringLiteral|autoCastToken
```
where
```js
type autoCastToken = (token: any)=> Expression|PartialStatement|NullLiteral;
```
So I've converted it to 
```js
SimpleReference|Identifier|RangeExpression|StringLiteral|Expression|PartialStatement|NullLiteral;
```
to avoid creating the autoCastToken type